robottelo/hosts.py

```diff
@@ -853,12 +855,19 @@ class Satellite(Capsule):
     def __init__(self, hostname=None, **kwargs):
         hostname = hostname or settings.server.hostname
         super().__init__(hostname=hostname, **kwargs)
-        self._init_nailgun()
-        self._init_cli()
-        self._init_airgun()
+        # create dummy classes for later population
+        self._api = type('api', (), {'_configured': False})
+        self._cli = type('cli', (), {'_configured': False})
+        self._ui_session = None
 
-    def _init_nailgun(self):
+    @property
+    def api(self):
         """Import all nailgun entities and wrap them under self.api"""
+        if not self._api:
+            self._api = type('api', (), {'_configured': False})
```

Comment:
You are constructing a type here. It is common to name the types with camel case. Would you mind to make it Api or API? Same goes for other use of `type()` in this PR

Reply:
While it is technically a class, we're not using it as such. Just an object to assign attributes to. With that in mind, I do not want to make it apparent that it is anything other than another attribute.